addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -841,467 +312,147 @@ def test_order_to_payment_currency(self):
-            }), (0, 0, {
-                'name': "OL/0002",
-                'product_id': self.product4.id,
-                'price_unit': 300,
-                'discount': 5.0,
-                'qty': 3.0,
-                'price_subtotal': 855,
-                'price_subtotal_incl': 855,
-            })],
-            'amount_tax': 855 * 2,
-            'amount_total': 855 * 2,
-            'amount_paid': 0.0,
-            'amount_return': 0.0,
-            'last_order_preparation_change': '{}'
-        })
-
-        # I click on the "Make Payment" wizard to pay the PoS order
-        context_make_payment = {"active_ids": [self.pos_order_pos1.id], "active_id": self.pos_order_pos1.id}
-        self.pos_make_payment = self.PosMakePayment.with_context(context_make_payment).create({
-            'amount': 855 * 2,
+        order, _ = self.create_backend_pos_order({
+            'order_data': {
+                'partner_id': self.partner_mobt.id,
+                'pricelist_id': self.partner_mobt.property_product_pricelist.id,
+            },
+            'line_data': [
+                {'product_id': self.ten_dollars_no_tax.product_variant_id.id},
+                {'product_id': self.twenty_dollars_no_tax.product_variant_id.id},
+            ],
+            'payment_data': [
+                {'payment_method_id': self.bank_payment_method.id, 'amount': 30},
+            ],
         })
-        # I click on the validate button to register the payment.
-        context_payment = {'active_id': self.pos_order_pos1.id}
-        self.pos_make_payment.with_context(context_payment).check()
+        self.assertEqual(order.state, 'paid', "Order should be in paid state.")
+        self.assertFalse(order.account_move, 'Invoice should not be attached to order yet.')
 
-        # I check that the order is marked as paid and there is no invoice
-        # attached to it
-        self.assertEqual(self.pos_order_pos1.state, 'paid', "Order should be in paid state.")
-        self.assertFalse(self.pos_order_pos1.account_move, 'Invoice should not be attached to order yet.')
-
-        # I generate an invoice from the order
-        res = self.pos_order_pos1.action_pos_order_invoice()
+        res = order.action_pos_order_invoice()
         self.assertIn('res_id', res, "No invoice created")
 
         # I test that the total of the attached invoice is correct
         invoice = self.env['account.move'].browse(res['res_id'])
         if invoice.state != 'posted':
             invoice.action_post()
-        self.assertAlmostEqual(
-            invoice.amount_total, self.pos_order_pos1.amount_total, places=2, msg="Invoice not correct")
+        self.assertAlmostEqual(invoice.amount_total, order.amount_total, places=2)
 
         for iline in invoice.invoice_line_ids:
             self.assertFalse(iline.tax_ids)
 
-        self.pos_config.current_session_id.action_pos_session_closing_control()
-
-    def test_order_with_deleted_tax(self):
```

Comment:
> - Test `test_order_with_deleted_tax` is removed because it is using
`sync_from_ui` method which should not be used in the backend.
Frontend tours are already testing its behavior

`test_order_with_deleted_tax` isn't about the use of `sync_from_ui`, it's about the fact that an order is created with some taxes, then, before closing the session, one of the taxes is removed. This test asserts such behaviour when closing the session. If we don't want to see `sync_from_ui` in the test, then we should replace this test with a tour.

Reply:
Ok I added it back